### PR TITLE
doc: Add missing doc

### LINKED ISSUE #36: Document every thing
GoOSe is missing a ton of documentation, including the out of date readme.

I will do this after #35 is merged

### PR DESCRIPTION
**Pull request by Skallwar**
Closes #36

### CODE REVIEW

**Review by CohenArthur — APPROVED** [review]

File: src/arch/mod.rs [hunk 1]
```diff
@@ -1,3 +1,9 @@
+//! This module implements "platform agnosticity". No matter what platform you're trying
```
Comment (Skallwar) [inline src/arch/mod.rs:1]:
```suggestion
//! This module implements "platform agnosticity" layer. No matter what platform you're trying
```

File: src/arch/mod.rs [hunk 2]
```diff
@@ -1,3 +1,9 @@
...
+//! to get goOSe working on, all of the conditions defined here must be met. On top of
```
Comment (Skallwar) [inline src/arch/mod.rs:2]: > all of the conditions defined here must be met

Which ones? I tough of making an arch trait but this will be in another pr with some refactor

File: src/arch/riscv64/mod.rs [hunk 3]
```diff
@@ -1,8 +1,12 @@
+//! RISCV64 module. Implements abstractions necessary to compile and use goOSe on
+//! RISCV64
```
Comment (Skallwar) [inline src/arch/riscv64/mod.rs:2]:
```suggestion
//! RISCV64 abstractions layer
```

File: src/arch/riscv64/mod.rs [hunk 4]
```diff
@@ -46,8 +50,9 @@ fn clear_bss() {
     println!("BSS cleared ({:#X} -> {:#X})", _bss_start, _bss_end);
 }
 
-/// Some architecture (x86...) have a specific instruction to write on some specific address (IO
-/// ports). RISCV don't so this is just a stub for writing at a specified address.
+/// Some architecture (x86...) have a specific instruction to write on some specific
+/// address (IO ports). RISCV does not, so this is just a stub for writing at
+/// a specified address.
```
Comment (Skallwar) [inline src/arch/riscv64/mod.rs:55]: What is wrong with my indentation? I use rustfmt, should be good I think, no?

File: src/main.rs [hunk 5]
```diff
@@ -22,10 +22,11 @@ mod serial;
 
 use cfg_if::cfg_if;
 
+#[doc(hidden)]
 static GREET: &str = "Talk to me, Goose !";
 
-/// After all architecture specific initialization for correct rust execution is done, this is "real"
-/// kernel entry point.
+/// After all architecture specific initialization for correct rust execution is done,
+/// this is the "real" kernel entry point.
```
Comment (Skallwar) [inline src/main.rs:29]: Same as above

File: src/panic/mod.rs [hunk 6]
```diff
@@ -1,10 +1,16 @@
...
+//! execution
+
 #[cfg(not(test))]
 use crate::println;
+#[cfg(not(test))]
+use core::panic::PanicInfo;
 
 #[cfg(not(test))]
 #[panic_handler]
+#[doc(hidden)]
```
Comment (Skallwar) [inline src/panic/mod.rs:13]: Why we don't want to document this?

File: src/arch/mod.rs (hunk 2 above)
Comment (CohenArthur) [reply src/arch/mod.rs:2]: For now it's just the external symbols, but I'm assuming that we'll have a trait and other conditions at some point

File: src/arch/riscv64/mod.rs (hunk 4 above)
Comment (CohenArthur) [reply src/arch/riscv64/mod.rs:54]: Oh nothing is wrong, I simply have a 90 characters limit and it was going over it. The indentation is fine!

File: src/panic/mod.rs (hunk 6 above)
Comment (CohenArthur) [reply src/panic/mod.rs:13]: Because we shouldn't call `panic` directly, only `panic!` which builds the `PanicInfo` argument

File: src/arch/mod.rs (hunk 2 above)
Comment (Skallwar) [reply src/arch/mod.rs:2]: Ok for me then

File: src/arch/riscv64/mod.rs (hunk 4 above)
Comment (Skallwar) [reply src/arch/riscv64/mod.rs:54]: Ok. I think rustfmt will break this line in the future so maybe just leave it as it is

File: src/panic/mod.rs (hunk 6 above)
Comment (Skallwar) [reply src/panic/mod.rs:13]: Ok

File: src/arch/riscv64/mod.rs (hunk 4 above)
Comment (Skallwar) [reply src/arch/riscv64/mod.rs:54]: My bad
